cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -309,10 +309,25 @@ func (c *Client) GetClusterVersions() (map[string]string, error) {
 
 // ClusterAvailable returns true if the cluster status indicates the cluster is available.
 func (c *Client) ClusterAvailable() (bool, error) {
-	_, err := c.GetClusterStatus()
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   c.Endpoints,
+		DialTimeout: 5 * time.Second,
+		TLS:         c.TLS,
+	})
 	if err != nil {
 		return false, err
 	}
+	defer cli.Close()
+
+	for _, ep := range c.Endpoints {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		_, err := cli.Status(ctx, ep)
+		cancel()
+		if err != nil {
+			return false, err
+		}
+		return true, nil
```

Comment:
i don't think this is correct. by using GetClusterStatus, the original code goes trough all the endpoints and checks for errors even if it ignores the actual responses.

/assign @fabriziopandini

Reply:
@neolit123 
Your comment has been addressed.